### PR TITLE
Feature/remove duplicate configurations

### PR DESCRIPTION
**Pull request by millerts**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ Please remove this after items are complete and/or moved into techdebt tickets.
 * ngBuild task in item-authoring-tool doesn't seem to know when it is up-to-date
 * PMD report is currently ~24000 lines just for common; start cleaning some of that up
     * remove rules that we don't care about
+* Remove @Configuration annotations from configuration classes in common
```
Comment (millerts) [inline CONTRIBUTING.md]: @malaffoon provided me this knowledge.  I thought Spring required the `@Configuration` annotation, but it is a stereotype similar to `@Service`.  I believe configuration classes in common are all manually imported now so we can remove the annotations.  That will allow this not to be required when doing package scanning.

File: common/src/main/java/org/opentestsystem/ap/common/config/ItemDataStoreManagerConfiguration.java [hunk 2]
```diff
@@ -58,7 +58,7 @@
 @EnableJpaRepositories(basePackages = "org.opentestsystem.ap.common.datastore.repository")
 @ConditionalOnProperty(prefix = "itembank.enabled", name = "database")
 @EnableConfigurationProperties({ItemBankProperties.class})
-public class DatabaseConfiguration {
+public class ItemDataStoreManagerConfiguration {
```
Comment (millerts) [inline common/src/main/java/org/opentestsystem/ap/common/config/ItemDataStoreManagerConfiguration.java]: I felt like this name is confusing.  This configuration is to set up all the Item related DataStoreManagers. Renamed to be clearer as one that does jdbc to unique tables doesn't require this service.

File: common/src/main/java/org/opentestsystem/ap/common/config/JdbcConfiguration.java [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 
 import javax.sql.DataSource;
 
 /**
  * This configuration is responsible for providing JDBC-required beans.
+ *
+ * The annotation is not on this class to avoid Spring's package scanning.  One can include this with the @Import
+ * configuration.
  */
-@Configuration
```
Comment (millerts) [inline common/src/main/java/org/opentestsystem/ap/common/config/JdbcConfiguration.java]: Moving this to common to reduce duplication.  This is/will be used by the following:

* test management service
* item migration
* tools for teachers service

File: common/src/main/java/org/opentestsystem/ap/common/config/YamlPropertiesConfiguration.java [hunk 4]
```diff
@@ -15,8 +15,10 @@
 
 /**
  * A {@link Configuration} class for loading SQL YML files as property source.
+ *
+ * The annotation is not on this class to avoid Spring's package scanning.  One can include this with the @Import
```
Comment (millerts) [inline common/src/main/java/org/opentestsystem/ap/common/config/YamlPropertiesConfiguration.java]: Moving this to common to reduce duplication.  This is/will be used by the following:

* test management service
* item migration
* tools for teachers service

File: item-bank-sync-service/src/main/java/org/opentestsystem/ap/itembank/sync/ItemBankSyncConfiguration.java [hunk 5]
```diff
@@ -64,7 +64,7 @@
 @Import({
     BrokerConfig.class,
     ContentNegotiationConfig.class,
-    DatabaseConfiguration.class,
+    ItemDataStoreManagerConfiguration.class,
```
Comment (millerts) [inline item-bank-sync-service/src/main/java/org/opentestsystem/ap/itembank/sync/ItemBankSyncConfiguration.java]: This is the rename.  Most of the other configuration file changes are like this one.

File: test-management-service/src/main/java/org/opentestsystem/ap/tms/configuration/TestManagementConfiguration.java [hunk 6]
```diff
@@ -25,11 +27,13 @@
 @Configuration
 @Import({
     ContentNegotiationConfig.class,
-    DatabaseConfiguration.class,
+    ItemDataStoreManagerConfiguration.class,
```
Comment (malaffoon) [inline test-management-service/src/main/java/org/opentestsystem/ap/tms/configuration/TestManagementConfiguration.java]: nit: not alphabetical anymore

File: tools-for-teachers-service/src/main/java/org/opentestsystem/ap/tft/configuration/ToolsForTeachersApplicationConfiguration.java [hunk 7]
```diff
@@ -14,7 +16,9 @@
     ContentNegotiationConfig.class,
     JwtWebSecurityConfiguration.class,
     ObjectMapperConfiguration.class,
-    StatusConfiguration.class
+    StatusConfiguration.class,
+    JdbcConfiguration.class,
```
Comment (malaffoon) [inline tools-for-teachers-service/src/main/java/org/opentestsystem/ap/tft/configuration/ToolsForTeachersApplicationConfiguration.java]: alphabetize?

**Pull request merged**